### ISSUE TITLE
browserify doesn't replace NODE_ENV

### ISSUE THREAD

**Issue by dwelle** [facebook/prop-types#267]
Unless I misunderstand, resulting bundle from browserify still contains some `process.env.NODE_ENV` references because the `prop-types` package depends on a another package that does not specify `browserify.transform` in its `package.json`. In particular, the `react-is` package (haven't checked if there are others).

```
npx browserify -r prop-types | grep NODE_ENV
```

The `browserify.transform` is a [non-global transform](https://github.com/browserify/browserify/issues/583), thus it doesn't transform dependencies.

**Comment by ljharb** [comment]
This is true; it's pretty important to apply envify at the app level. I'm not sure how else to address this issue.

The only other dep is `object-assign` and there shouldn't be anything env-specific in there.

**Comment by dwelle** [comment]
Yea, one way is for the user to run custom global transform like:
```
stream.transform(envify(), { global: true })

**Comment by ljharb** [comment]
Typically bundles are where the distinction matters, and bundlers *do* typically replace it in the entire bundle - it's not much of a problem for node.

Perhaps you could file an issue on the react repo for `react-is`, and then this one can be closed?

**Comment by oliviertassinari** [comment]
It feels like removing the dependency on loose-envify would make sense. Referencing: https://github.com/facebook/react/issues/27748 where this is discussed.
